### DISCUSSION TITLE
Custom color for all Write-Log function Output

### DISCUSSION THREAD

**Discussion by retro-encabulator** [PSAppDeployToolkit/PSAppDeployToolkit#755] [Ideas]
Currently the Write-Log function allows coloring of the text output by assigning -Severity in the script. It would be convenient to be able to set a default color for **all** Write-Log function output to distinguish it from the rest of the native PSADTK log output. This setting could be managed via the AppDeployToolkit.xml file.

**Comment by That-Annoying-Guy** [comment]
I have done this already in my personal fork about 7 years ago.
What was hard was picking colors that worked **on the screen.**
It will NOT work with CMTrace formatted logs. (IOW: No new colors showing in CMTrace.exe)
NOTE: I don't use  CMTrace formatted logs
I'm able to display (4) grey and (5) green **on the screen.**

I can post my modified Write-log function in the [Extensions ](https://discourse.psappdeploytoolkit.com/c/thetoolkit/extensions/13 ) section. It's from the 3.8.5 beta so it might not work with 3.9.x.

**Comment by mjr4077au** [comment]
The reality is no one ever sees the console output unless you're debugging the script. We have -Severity 0 for green, 2 for yellow, and 3 for error, which I think is sufficient for most, if not all needs.
